api/src/data/migrations/023.create-incident-logs.ts

```diff
@@ -0,0 +1,43 @@
+import * as knex from "knex";
+
+export async function up(knex: knex.Knex) {
+    await knex.schema.createTable("incident_logs", function(table) {
+        table.increments("id").primary().notNullable();
+        table.integer("incident_id").notNullable();
+        table.string("old_description", 4096).notNullable();
+        table.string("old_sensitivity", 256).nullable();
+        table.string("old_supervisor", 8).nullable();
+        table.datetime("old_created_date").notNullable();
+        table.string("old_status", 8).nullable();
+        table.integer("old_incident_type_id").nullable();
+        table.string("new_description", 4096).notNullable();
+        table.string("new_supervisor", 8).nullable();
+        table.string("new_sensitivity", 8).nullable();
+        table.datetime("new_created_date").notNullable();
+        table.string("new_status", 8).nullable();
+        table.integer("new_incident_type_id").nullable();
+        table.string("changer_employee_id", 256).nullable();
+        table.integer("changer_role_id").nullable();
+        table.datetime("changed_date").notNullable();
+        table.string("log_comment", 4096).notNullable();
+        table.string("user_action", 8).notNullable();
+
+        table.foreign("incident_id").references("incidents.id");
+        table.foreign("old_sensitivity").references("sensitivities.code");
+        // old_status foreign keys. Why are there two??
+        table.foreign("old_status", "incident_log_old_incident_status_code_fk").references("incident_statuses.code");
+        // shouldn't this be incident_log_old_incident_status_code_fk2 ?
+        table.foreign("old_status", "incident_log_old_status_code_fk2").references("incident_statuses.code");
```

Comment:
Same comment as previously.

Also if you want to leave a comment that should be actioned prefix it with `TODO`. Many  editors support highlight lines starting with TODO.